### ISSUE TITLE
WebSocket event listeners not called if table is originally empty

### ISSUE THREAD

**Issue by jc5826** [balderdashy/sails#4331]
**Sails version**: 1.0.0-46
**Node version**: v8.9.4
**NPM version**: 5.7.1
**DB adapter name**: N/A
**DB adapter version**: N/A
**Operating system**: macOS 10.13.3

<hr/>

I'm new to Sails and to node in general, but I have spent a lot time trying to figure out why WebSocket messages via the ORM's (model) publish method are not reaching the client.  It appears that publish events are not triggered for tables if they are originally empty.  If the table has at least one record, the client event listener is called.  As you might imagine, since most of the tables are not empty it took me some time to realize what was going on.

client code:

```
angular.module('Test', []).controller('BaseCtrl', function($scope) {
    io.socket.get('/walkin', function(result) {
        console.log('walkin', result);
    });

    io.socket.on('walkin', function(event) {
        // this does not get called if the 'walkin' table is empty
        console.log('walkin event', event);
    });

    $scope.createWalkin = function() {
        var data = {walkin_id: '03421d52f8160500:211', session_id: 211, visitor_id: '03421d52f8160500'};
        io.socket.post('/addWalkin', data, function(resData, jwRes) {
            jwRes.statusCode; // => 200
        });
    };
});
```

server code:
```

module.exports = {

    addWalkin: function(req, res) {
        var data = req.body;
        console.log('data', data);
        Walkin.findOrCreate({walkin_id: data.walkin_id}, data).exec(function(err) {
            if (err) return console.log('error', err);
            Walkin.publish([data.walkin_id], {data: data, verb: 'created', id: data.walkin_id});
        });
        res.json({done: 1});
    }

};
```

**Comment by sailsbot** [comment]
@jc5826 Thanks for posting, we'll take a look as soon as possible.

---

For help with questions about Sails, [click here](http://sailsjs.com/support).  If you&rsquo;re interested in hiring @sailsbot and her minions in Austin, [click here](http://sailsjs.com/studio).

**Comment by jc5826** [comment]
I believe I figured out why this is. What the io.socket.get('/foo') does is automate the process of subscribing to each of the records that are originally in the table when the request is made. The io.socket.on('foo') listener will only receive messages for *only* those records. 

As many times as I viewed videos and read the documentation, this was not clear to me. IMO the documentation can be improved here.

Turns out I was able to accomplish what I wanted (subscribe to all existing and future records) by using sails.sockets.broadcast() and sails.sockets.join() directly instead of relying on sails.js magic.
